lib/interactivity-api/class-wp-interactivity-api-full-page-navigation.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Enables full page client-side navigation for the Interactivity API.
+ *
+ * IMPORTANT: this code is not meant to be included in Gutenberg yet. Instead,
+ * it will be part of an external plugin to test this feature separately.
+ */
+
+if ( ! class_exists( 'WP_Interactivity_API_Full_Page_Navigation' ) ) {
+
+	/**
+	 * Class to enable full page client-side navigation for the Interactivity API.
+	 */
+	class WP_Interactivity_API_Full_Page_Navigation {
+
+		private static $instance = null;
+
+		public function __construct() {
+			add_action( 'init', array( $this, 'set_default_mode' ), 9 );
+			add_action( 'wp_head', array( $this, 'buffer_start' ) );
+			add_action( 'wp_footer', array( $this, 'buffer_end' ) );
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_script_modules' ) );
+		}
+
+		/**
+		 * Returns whether the client navigation mode is `experimentalFullPage`.
+		 */
+		public function is_enabled() {
+			$iapi_router_config     = wp_interactivity_config( 'core/router' );
+			$client_navigation_mode = $iapi_router_config['clientNavigationMode'];
+			return 'experimentalFullPage' === $client_navigation_mode;
+		}
```

Comment:
This should avoid a warning when the `clientNavigationMode` index doesn't exist.

```suggestion
		public function is_enabled() {
			$iapi_router_config = wp_interactivity_config( 'core/router' );
			return 'experimentalFullPage' === $iapi_router_config['clientNavigationMode'] ?? '';
		}
```

Reply:
Done at https://github.com/woocommerce/gutenberg/pull/13/commits/d112a779bb5cb134bc83f3126e997e94664803ea.